tests/components/mqtt/test_select.py

```diff
@@ -74,112 +73,108 @@ def select_platform_only():
         yield
 
 
-async def test_run_select_setup(
-    hass: HomeAssistant, mqtt_mock_entry_with_yaml_config: MqttMockHAClientGenerator
-) -> None:
-    """Test that it fetches the given payload."""
-    topic = "test/select"
-    await async_setup_component(
-        hass,
-        mqtt.DOMAIN,
+@pytest.mark.parametrize(
+    "hass_config",
+    [
         {
             mqtt.DOMAIN: {
                 select.DOMAIN: {
-                    "state_topic": topic,
-                    "command_topic": topic,
+                    "state_topic": "test/select_stat",
+                    "command_topic": "test/select_cmd",
                     "name": "Test Select",
                     "options": ["milk", "beer"],
                 }
             }
-        },
-    )
-    await hass.async_block_till_done()
-    await mqtt_mock_entry_with_yaml_config()
+        }
+    ],
+)
+async def test_run_select_setup(
+    hass: HomeAssistant, mqtt_mock_entry_no_yaml_config: MqttMockHAClientGenerator
+) -> None:
+    """Test that it fetches the given payload."""
+    await mqtt_mock_entry_no_yaml_config()
 
-    async_fire_mqtt_message(hass, topic, "milk")
+    async_fire_mqtt_message(hass, "test/select_stat", "milk")
```

Comment:
This is not so nice, you can inject this as a parametrization `topic` and use that both here and to generate the config.